Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -318,12 +339,43 @@ namespace mu2e {
       kkbf_ = std::move(std::make_unique<KKBField>(*bfmgr,*det));
     }
     if(print_ > 0) kkbf_->print(std::cout);
+
+    // Print the fit direction configuration
+    if(print_ > -1) printf("[LoopHelixFit::%s::%s] Fit dz/dt direction = %.0f, PDG = %i, use helix slope = %o with threshold %.1f, use helix slope cut = %o with theshold %.1f\n",
+                           __func__, moduleDescription().moduleLabel().c_str(), fdir_.dzdt(), (int) fpart_,
+                           useHelixSlope_, (useHelixSlope_) ? slopeSigThreshold_ : -1.f,
+                           useSlopeSigCut_, (useSlopeSigCut_) ? slopeSigCut_ : -999.f);
   }
 
-  void LoopHelixFit::produce(art::Event& event ) {
-    // empty collections
-    static MEASCOL nohits; // empty
-    static EXINGCOL noexings; // empty
+  HelixRecoDir::PropDir LoopHelixFit::chooseHelixDir(HelixSeed const& hseed) const {
+    if(useHelixSlope_) return hseed.recoDir().predictDirection(slopeSigThreshold_);
+    if(fdir_.fitDirection() == TrkFitDirection::FitDirection::upstream  ) return HelixRecoDir::PropDir::upstream;
+    if(fdir_.fitDirection() == TrkFitDirection::FitDirection::downstream) return HelixRecoDir::PropDir::downstream;
+    return HelixRecoDir::PropDir::ambiguous;
+  }
+
+  bool LoopHelixFit::compare_tracks(KKTRK const& ktrk_1, KKTRK const& ktrk_2) const {
+    // return true to select the first track, false to select the second
+
+    const bool goodfit_1(goodFit(ktrk_1)), goodfit_2(goodFit(ktrk_2));
+    if(print_>1) printf(" [LoopHelixFit::%s]: status 1 = %o, status 2 = %o\n", __func__, goodfit_1, goodfit_2);
+
+    if(!goodfit_2) return true; //if the second is bad or if both are bad, default to the first
+    if(!goodfit_1) return false; //if only the first is bad, accept the second
+
+    const auto p_1(ktrk_1.fitStatus().chisq_.probability());
+    const auto p_2(ktrk_2.fitStatus().chisq_.probability());
+    const auto n_calo_1(ktrk_1.caloHits().size());
```

Comment:
The indenting looks odd here, please check

Reply:
I'm not sure if I can see the issue, it looks like it matches the other functions' indenting.